papers/Kilpatrick2024_Alopeke/Analysis/py/alopeke_utils2.py

```diff
@@ -0,0 +1,130 @@
+import sys, os
+import numpy as np
+from scipy.stats import norm
+
+from astropy.table import Table
+
+sys.path.append(os.path.abspath("../Analysis/py"))
+import alopeke_defs
+
+def calc_zeropoint(data_dict, camera:str):
+
+    # Instrumental magnitude for star 1 and star 2
+    m_star1 = -2.5 * np.log10(data_dict['C_star1_full']-data_dict['C_bkg_full'])
+    m_star2 = -2.5 * np.log10(data_dict['C_star2_full']-data_dict['C_bkg_full'])
+
+    # Uncertainties on instrumental magnitudes - assume Poisson
+    merr_star1 = 1.086 * np.sqrt(data_dict['C_star1_full'])/data_dict['C_star1_full']
+    merr_star2 = 1.086 * np.sqrt(data_dict['C_star1_full'])/data_dict['C_star1_full']
+
+    # Now get zero points
+    zpt_star1 = 0.
+    zpt_star2 = 0.
+    if camera=='red':
+        zpt_star1 = alopeke_defs.i_1 - m_star1
+        zpt_star2 = alopeke_defs.i_2 - m_star2
+    elif camera=='blue':
+        zpt_star1 = alopeke_defs.r_1 - m_star1
+        zpt_star2 = alopeke_defs.r_2 - m_star2
+    else:
+        raise Exception(f'ERROR: Unrecognized camera {camera}')
+
+    zpt = (merr_star1 * zpt_star1 + merr_star2 * zpt_star2)/(merr_star1 + merr_star2)
+    zpt_err = 1./np.sqrt(2) * np.sqrt(merr_star1**2 + merr_star2**2)
+
+    return(zpt, zpt_err)
+
+def get_star_flux(date:str, data_dict):
+    mask = np.abs(data_dict['MJD_star1_full']-alopeke_defs.FRB_time[date].mjd)*24*3600<0.163
```

Comment:
could the 0.163 be read from a configuration file rather than being hardcoded? @charlie_kilpatrick

Reply:
I've updated to `...alopeke_defs.ata.to('s').value` instead of the hard coded value.